### PR TITLE
Testing mode and Result links support

### LINKED ISSUE #17: Add Link to see results in Scope
We need to add the link to see the results in scope after the agent finish.

### LINKED ISSUE #22: Support SCOPE_TESTING_MODE env var (w/ IN_CI fallback)

### LINKED ISSUE #30: Flush buffer in prod mode when having data without waiting the timeout

### PR DESCRIPTION
**Pull request by tonyredondo**
Closes #17 
Closes #22 
Closes #30

This PR adds support for `SCOPE_TESTING_MODE` and Scope results link

### CODE REVIEW

File: recorder.go [hunk 1]
```diff
@@ -142,5 +157,11 @@ func (r *SpanRecorder) SendSpans() error {
 		r.spans = nil
 	}
 
+	if resp.StatusCode < 400 {
```
Comment (fermayo) [inline recorder.go]: We should retry on 5xx errors, and give up the payload on 4xx error

File: recorder.go [hunk 2]
```diff
@@ -38,7 +47,7 @@ func (r *SpanRecorder) RecordSpan(span tracer.RawSpan) {
 }
 
 func (r *SpanRecorder) loop() error {
-	ticker := time.NewTicker(1 * time.Second)
+	ticker := time.NewTicker(r.flushFrequency)
```
Comment (fermayo) [inline recorder.go]: This means that in non-testing mode, spans in the buffer will wait for up to a minute to be flushed. This cannot happen - we have to flush every second, but only send empty payloads every minute if testing mode is disabled

File: recorder.go (hunk 1 above)
Comment (tonyredondo) [reply recorder.go]: Ok, I added a retry algorithm

File: recorder.go (hunk 2 above)
Comment (tonyredondo) [reply recorder.go]: Ok, done

**Pull request merged**